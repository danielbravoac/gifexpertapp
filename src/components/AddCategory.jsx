import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputvalue, setInputValue] = useState("");
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    const newInputValue = inputvalue.trim();

    if (newInputValue.length === 0) return;

    onNewCategory(newInputValue);
    setInputValue("");
  };
  return (
    <>
      <form onSubmit={onSubmit} aria-label="form">
        <input
          type="text"
          placeholder="Buscar gifs"
          value={inputvalue}
          onChange={onInputChange}
          className="w-full p-2 border border-gray-300 rounded-md"
        />
      </form>
    </>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
