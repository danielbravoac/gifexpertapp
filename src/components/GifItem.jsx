import PropTypes from "prop-types";
export const GifItem = ({ title, url, id }) => {
  return (
    <div className=" bg-white shadow-lg rounded-md  flex flex-col items-center overflow-hidden h-96">
      <img
        src={url}
        alt={title}
        className="object-cover object-center  w-full flex-1"
        aria-label="gif"
      />
      {/* <p className="p-4 ">{title != " " ? title : "[Gif Sin Nombre]"}</p> */}
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
/* GifItem.defaultProps = {
  title: "[Gif Sin Nombre]",
  url: "",
}; */
