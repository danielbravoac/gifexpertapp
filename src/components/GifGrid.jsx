import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const { images, isLoading, isError } = useFetchGifs(category);
  return (
    <div className="space-y-5">
      <h3 className="font-bold text-4xl text-blue-400">
        {category.charAt(0).toUpperCase() + category.slice(1)}
      </h3>
      {isLoading && (
        <h2 className="bg-blue-200 text-blue-800 rounded-lg animate-pulse p-4">
          Cargando...
        </h2>
      )}
      {isError && (
        <h2 className="bg-red-200 text-red-800 p-4 rounded-lg animate-pulse">
          Estamos teniendo problemas para cargar los gifs
        </h2>
      )}
      <div className="grid lg:grid-cols-4 md:grid-cols-3 gap-2">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </div>
  );
};
