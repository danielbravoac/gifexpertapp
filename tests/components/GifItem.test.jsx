import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas de <GifItem/>", () => {
  const title = "Un titulo";
  const imgUrl = "https://localhost/algo.jpg";
  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={imgUrl} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar una imagen", () => {
    render(<GifItem title={title} url={imgUrl} />);
    const img = screen.getByLabelText("gif");
    //console.log(img);
    expect(img).toBeTruthy();
  });
  test("debe de mostrar la imagen con la url y alt indicado", () => {
    render(<GifItem title={title} url={imgUrl} />);
    //screen.debug();
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(imgUrl);
    expect(alt).toBe(title);
  });
});
