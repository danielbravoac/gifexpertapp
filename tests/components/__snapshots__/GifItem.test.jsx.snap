// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class=" bg-white shadow-lg rounded-md  flex flex-col items-center overflow-hidden h-96"
  >
    <img
      alt="Un titulo"
      aria-label="gif"
      class="object-cover object-center  w-full flex-1"
      src="https://localhost/algo.jpg"
    />
  </div>
</div>
`;
